packages/common/src/Adaptor.js

```diff
@@ -586,62 +598,34 @@ export function parseCsv(csvData, parsingOptions = {}, callback) {
     throw new Error('chunkSize must be at least 1');
   }
 
-  return state => {
-    return new Promise((resolve, reject) => {
-      let buffer = [];
-
-      const parser =
-        typeof csvData === 'string'
-          ? parse(csvData, options)
-          : csvData.pipe(parse(options));
+  return async state => {
+    let buffer = [];
 
-      const flushBuffer = currentState => {
-        const nextState = callback
-          ? callback(currentState, buffer)
-          : composeNextState(currentState, buffer);
+    const parser =
+      typeof csvData === 'string'
+        ? parse(csvData, options)
+        : csvData.pipe(parse(options));
 
-        buffer = [];
+    const flushBuffer = async currentState => {
+      const nextState = callback
+        ? await callback(currentState, buffer)
+        : composeNextState(currentState, buffer);
 
-        return [nextState, buffer];
-      };
+      buffer = [];
 
-      parser.on('readable', function () {
-        let chunk;
+      return [nextState, buffer];
+    };
 
-        while ((chunk = parser.read()) !== null) {
-          buffer.push(chunk);
-
-          if (buffer.length >= options.chunkSize) {
-            const [nextState, nextBuffer] = flushBuffer(state);
-            // eslint-disable-next-line no-param-reassign
-            state = nextState;
-            buffer = nextBuffer;
-          }
-        }
-      });
-      parser.on('error', function (error) {
-        reject(error);
-      });
-
-      parser.on('end', function () {
-        const [finalState] = flushBuffer(state);
-        resolve(finalState);
-      });
-    });
+    for await (const record of parser) {
+      buffer.push(record);
+      if (buffer.length === options.chunkSize) {
+        const [nextState, nextBuffer] = await flushBuffer(state);
+        // eslint-disable-next-line no-param-reassign
+        state = nextState;
+        buffer = nextBuffer;
+      }
+    }
+    const [finalState] = await flushBuffer(state);
+    return finalState;
   };
 }
-// /**
```

Comment:
Hmm, this shouldn't be removed here either.

Is this diff just out of date? If we're behind then that's ok :)